src/rmc/tests.rs

```diff
@@ -0,0 +1,255 @@
+    future::{self, BoxFuture, FutureExt},
+    stream::{self, Stream},
+};
+use rand::Rng;
+use std::{pin::Pin, time::Duration};
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash, Ord, PartialOrd)]
+struct Hash {
+    byte: u8,
+}
+
+impl Signable for Hash {
+    type Hash = [u8; 1];
+
+    fn hash(&self) -> Self::Hash {
+        [self.byte]
+    }
+}
+
+type TestMessage = Message<Hash, TestSignature, TestPartialMultisignature>;
+
+struct TestNetwork {
+    outgoing_rx: Pin<Box<dyn Stream<Item = (NodeIndex, TestMessage)>>>,
+    incoming_txs: Vec<UnboundedSender<TestMessage>>,
+    message_filter: Box<dyn FnMut(NodeIndex, TestMessage) -> bool>,
+}
+
+impl TestNetwork {
+    fn new(
+        node_count: NodeCount,
+        message_filter: impl FnMut(NodeIndex, TestMessage) -> bool + 'static,
+    ) -> (
+        Self,
+        Vec<(
+            UnboundedReceiver<TestMessage>,
+            UnboundedSender<(NodeIndex, TestMessage)>,
+        )>,
+    ) {
+        let all_nodes: Vec<_> = (0..node_count.0).map(NodeIndex).collect();
+        let (incomng_txs, incoming_rxs): (Vec<_>, Vec<_>) =
+            all_nodes.iter().map(|_| unbounded::<TestMessage>()).unzip();
+        let (outgoing_txs, outgoing_rxs): (Vec<_>, Vec<_>) = {
+            all_nodes
+                .iter()
+                .map(|_| {
+                    let (tx, rx) = unbounded::<(NodeIndex, TestMessage)>();
+                    (tx, rx)
+                })
+                .unzip()
+        };
+        let network = TestNetwork {
+            outgoing_rx: Box::pin(stream::select_all(outgoing_rxs)),
+            incoming_txs: incomng_txs,
+            message_filter: Box::new(message_filter),
+        };
+
+        let channels = incoming_rxs.into_iter().zip(outgoing_txs).collect();
+        (network, channels)
+    }
+
+    fn broadcast_message(&mut self, msg: TestMessage) {
+        for tx in &mut self.incoming_txs {
+            tx.unbounded_send(msg.clone()).unwrap();
+        }
+    }
+}
+
+impl TestNetwork {
+    async fn run(&mut self) {
+        while let Some((recipient, message)) = self.outgoing_rx.next().await {
+            if (self.message_filter)(recipient, message.clone()) {
+                self.incoming_txs[recipient.0]
+                    .unbounded_send(message)
+                    .unwrap();
+            }
+        }
+    }
+}
+
+fn prepare_keychains(node_count: NodeCount) -> Vec<TestMultiKeychain> {
+    (0..node_count.0)
+        .map(|i| TestMultiKeychain::new(node_count, i.into()))
+        .collect()
+}
+
+struct TestData<'a> {
+    network: TestNetwork,
+    rmcs: Vec<ReliableMulticast<'a, Hash, TestMultiKeychain>>,
+}
+
+impl<'a> TestData<'a> {
+    fn new(
+        node_count: NodeCount,
+        keychains: &'a [TestMultiKeychain],
+        message_filter: impl FnMut(NodeIndex, TestMessage) -> bool + 'static,
+    ) -> Self {
+        let (network, channels) = TestNetwork::new(node_count, message_filter);
+        let mut rmcs = vec![];
```

Comment:
`Vec::new()` is probably mode idiomatic for empty vectors.

Reply:
Didn't know that, done `s/vec![]/Vec::new()/g`.